### ISSUE TITLE
Unsupported architecture / different architectures in Carthage/Build/IOS than the original repo

### ISSUE THREAD

**Issue by christinatealium** [Carthage/Carthage#2764]
* carthage install method: [ ] .pkg, [ ] homebrew, [ ] source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 0.33.0
* `xcodebuild -version`: Xcode 10.2 Build version 10E125
* Are you using `--no-build`? 
* Are you using `--no-use-binaries`? 
* Are you using `--use-submodules`? I did, yes.
* Are you using `--cache-builds`? 
* Are you using `--new-resolver`? 

**Cartfile**
```
binary "https://tags.tiqcdn.com/dle/tealiummobile/tealium-ios-carthage/tealium-carthage.json"
```

**Carthage Output**
When running: `lipo -info TealiumIOSLifecycle.framework/TealiumIOSLifecycle`:
```
Architectures in the fat file: TealiumIOSLifecycle.framework/TealiumIOSLifecycle are: armv7 arm64
```

**Actual outcome**
When running `lipo -info TealiumIOSLifecycle.framework/TealiumIOSLifecycle` in the latest release.zip from our repo:
```
Architectures in the fat file: TealiumIOSLifecycle.framework/TealiumIOSLifecycle are: i386 x86_64 armv7 arm64
```

**Expected outcome**
I am wondering why the architectures are not the same as directly from our repo. Anything additional needed?

Someone else was having a similar problem and posted an issue on our GitHub page. Not positive it is related, but thought I would include it: https://github.com/Tealium/tealium-ios/issues/99

I tried to look at the code in Carthage and didn't see anything obvious. Thanks for the help!

**Comment by mdiep** [comment]
> **Carthage Output**
> When running: `lipo -info TealiumIOSLifecycle.framework/TealiumIOSLifecycle`:

Where are you getting the `.framework` from on disk when running this test?

**Comment by christinatealium** [comment]
Carthage version:
<Project Folder>/Carthage/Build/iOS

Version direct from our repo:
Folder containing download from repo

**Comment by blender** [comment]
You are pointing carthage to a prebuilt binary.

`binary "https://tags.tiqcdn.com/dle/tealiummobile/tealium-ios-carthage/tealium-carthage.json"`

I suppose those architectures are not included in the binary that is being downloaded.

**Comment by blender** [comment]
```
$ carthage update --platform iOS
*** Downloading binary-only framework tealium-carthage at "https://tags.tiqcdn.com/dle/tealiummobile/tealium-ios-carthage/tealium-carthage.json"
Failed to read file or folder at /dle/tealiummobile/tealium-ios-carthage/tealium-carthage.json: Error Domain=Result.AnyError Code=1 "Could not connect to the server."
```

curl, same result
```
$ curl https://tags.tiqcdn.com/dle/tealiummobile/tealium-ios-carthage/tealium-carthage.json
curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to tags.tiqcdn.com:443
```

So I created my own binary definition `tealium.json`
```
{
 "5.4.11": "https://github.com/Tealium/tealium-ios/archive/5.4.11.zip"
}
```

and the following Cartfile

```
binary "tealium.json"
```

Run carthage
```
$ carthage update --platform iOS
*** Downloading binary-only framework tealium at "file:///Users/Tommaso/Code/lab/Carthage-2764/tealium.json"
*** Downloading binary-only framework tealium at "file:///Users/Tommaso/Code/lab/Carthage-2764/tealium.json"
*** xcodebuild output can be found in /var/folders/b4/qz7m1r9x3mj1bm3nfsjjf64m0000gp/T/carthage-xcodebuild.tF8ZeN.log
*** Downloading tealium.framework binary at "5.4.11"
$ file Carthage/Build/iOS/TealiumIOS.framework/TealiumIOS
Carthage/Build/iOS/TealiumIOS.framework/TealiumIOS: Mach-O universal binary with 4 architectures: [i386:Mach-O dynamically linked shared library i386] [arm64]
Carthage/Build/iOS/TealiumIOS.framework/TealiumIOS (for architecture i386):	Mach-O dynamically linked shared library i386
Carthage/Build/iOS/TealiumIOS.framework/TealiumIOS (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
Carthage/Build/iOS/TealiumIOS.framework/TealiumIOS (for architecture armv7):	Mach-O dynamically linked shared library arm_v7
Carthage/Build/iOS/TealiumIOS.framework/TealiumIOS (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

but 

```
file Carthage/Build/iOS/TealiumIOSLifecycle.framework/TealiumIOSLifecycle
Carthage/Build/iOS/TealiumIOSLifecycle.framework/TealiumIOSLifecycle: Mach-O universal binary with 2 architectures: [arm_v7:Mach-O dynamically linked shared library arm_v7] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
Carthage/Build/iOS/TealiumIOSLifecycle.framework/TealiumIOSLifecycle (for architecture armv7):	Mach-O dynamically linked shared library arm_v7
Carthage/Build/iOS/TealiumIOSLifecycle.framework/TealiumIOSLifecycle (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

which is different form the one manually downloaded from the zip at the same location:

```
$ file ~/Downloads/tealium-ios-5.4.11/TealiumIOSLifecycle.framework/TealiumIOSLifecycle
/Users/Tommaso/Downloads/tealium-ios-5.4.11/TealiumIOSLifecycle.framework/TealiumIOSLifecycle: Mach-O universal binary with 4 architectures: [i386:Mach-O dynamically linked shared library i386] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
/Users/Tommaso/Downloads/tealium-ios-5.4.11/TealiumIOSLifecycle.framework/TealiumIOSLifecycle (for architecture i386):	Mach-O dynamically linked shared library i386
/Users/Tommaso/Downloads/tealium-ios-5.4.11/TealiumIOSLifecycle.framework/TealiumIOSLifecycle (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
/Users/Tommaso/Downloads/tealium-ios-5.4.11/TealiumIOSLifecycle.framework/TealiumIOSLifecycle (for architecture armv7):	Mach-O dynamically linked shared library arm_v7
/Users/Tommaso/Downloads/tealium-ios-5.4.11/TealiumIOSLifecycle.framework/TealiumIOSLifecycle (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

**Comment by christinatealium** [comment]
Thanks @blender, are you saying there is a problem with our binary in the .zip? Im not super familiar with the carthage process, to be honest. If this is something we can change on our end let me know.

What you were experiencing in the post just before this was exactly what I was enquiring about. The version of the framework that is in the Carthage/Build/iOS folder is different that the same framework we are including in the zip.

How do I prevent this?

Thanks again for looking at this!

**Comment by blender** [comment]
I am unable to locate the precise issue as am I encountering another issue while performing the update. I think the two are related.

**Comment by christinatealium** [comment]
Got it, thanks again for checking.

@mdiep I don't suppose you have any ideas?

**Comment by mdiep** [comment]
No, I'm not sure why this would be the case. 😕

**Comment by blender** [comment]
@christinatealium in the zip you are pointing to there are frameworks with the exact same names in a directory called "DevicesOnly". Carthage is overwriting the Frameworks are top level with these ones.

In an archive there can't be multiple frameworks with the same name.

**Comment by blender** [comment]
This BTW causes a race between copying frameworks to the destination and reading dwarfdumps uuids, since the copy from one position has finished but the overwrite from another position in progress.

@mdiep  I think carthage should exit right away if it finds multiple frameworks with the same name in an archive.

**Comment by blender** [comment]
closed by #2765
